source/patterns/@aws-solutions-constructs/aws-fargate-stepfunctions/lib/index.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Note: To ensure CDKv2 compatibility, keep the import statement for Construct separate
+import { Construct } from "@aws-cdk/core";
+import * as defaults from "@aws-solutions-constructs/core";
+import * as ecs from "@aws-cdk/aws-ecs";
+import * as ec2 from "@aws-cdk/aws-ec2";
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as logs from '@aws-cdk/aws-logs';
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+
+export interface FargateToStepfunctionsProps {
+  /**
+   * Whether the construct is deploying a private or public API. This has implications for the VPC deployed
+   * by this construct.
+   *
+   * @default - none
+   */
+  readonly publicApi: boolean;
+  /**
+   * Optional custom properties for a VPC the construct will create. This VPC will
+   * be used by the new Fargate service the construct creates (that's
+   * why targetGroupProps can't include a VPC). Providing
+   * both this and existingVpc is an error. A Step Functions Interface
+   * endpoint will be included in this VPC.
+   *
+   * @default - none
```

Comment:
Actually, there is. If this and existingVpc are both undefined, this uses DefaultPublicPrivateVpcProps() or DefaultIsolatedVpcProps() (depending on publicApi) from vpc-defaults.ts. Let's say:

@default A set of defaults from vpc-defaults.ts: DefaultPublicPrivateVpcProps() for public APIs and DefaultIsolatedVpcProps() for private APIs.

Reply:
(your comment on my PR made me realize this). Let's try to do this going forward, we can discuss updating existing construct comments in the future.